doc/md/fundamentals/7-messaging.md

```diff
@@ -0,0 +1,170 @@
+---
+sidebar_position: 7
+---
+
+# Messaging
+
+## Messaging restrictions
+
+ICP enforces rules on when and how [canisters](https://internetcomputer.org/docs/building-apps/essentials/canisters) communicate. Motoko has static (compile-time) messaging restrictions to prevent certain execution errors.
+
+| Restriction | Reason |
+|-------------|--------|
+| A canister cannot send messages during installation. | Prevents execution errors at deployment. |
+| A query function cannot send messages. | Query functions run locally and do not trigger updates. |
+| Shared functions cannot be called in a synchronous context. | Shared calls require `async` execution. |
+| Only async contexts support [error handling](https://internetcomputer.org/docs/motoko/fundamentals/error-handling) (`try/catch`). | Messaging errors only occur in async contexts. |
+
+In Motoko, an expression is in an async context if it appears in an `async` function. Query functions are 'ready only' functions, so they do not create an async context and cannot use `await` or send messages.
+
+```motoko no-repl
+actor Counter {
+    stable var count : Nat = 0;
+
+    public shared func increment() : async () {
```

Comment:
```suggestion
persistent actor Counter {
    var count : Nat = 0;

    public func increment() : async () {
```

Reply:
I'd turn every(\*) actor to an `persistent actor` as that's our current default and a standard we are recommending.
(\*) Unless there is educational value in doing it otherwise, e.g. when introducing `persistent`/`stable` etc.